packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/mtom/parser.ts

```diff
@@ -1,159 +1,90 @@
-      acc[key] = value;
-    }
-    return acc;
-  }, {});
-
-  if (!headers["content-id"]) {
-    throw new Error("No Content-ID header found in headers.");
-  }
-  if (!headers["content-type"]) {
-    throw new Error("No Content-Type header found in headers.");
-  }
-  return {
-    ContentID: stripTags(headers["content-id"]),
-    ContentTransferEncoding: headers["content-transfer-encoding"],
-    ContentType: headers["content-type"],
-  };
+export interface IMTOMAttachments {
+  parts: IMTOMPart[];
 }
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-function containsMultipartCidReference(documentResponse: any): boolean {
-  return !!documentResponse.Document?.Include?._href;
-}
-
-export function parseMtomResponse(mtomMessage: string, contentType: string): DocumentResponse[] {
-  const contentTypeParams = parseMtomContentType(contentType);
-
-  const boundary = `--${contentTypeParams.boundary}`;
-  const parts = mtomMessage.split(boundary).slice(1, -1);
-
-  const documentResponsesMultipart: DocumentResponse[] = [];
-  const documentResponsesRegular: DocumentResponse[] = [];
-  const attachments: Record<string, string> = {};
-
-  parts.forEach(part => {
-    let splitter = carriageReturnLineFeed;
-    let headersEndIndex = -1;
-    if (contentTypeParams.startInfo) {
-      splitter = contentTypeParams.startInfo + splitter;
-      headersEndIndex = part.indexOf(splitter);
-    }
-    if (headersEndIndex < 0) {
-      splitter = carriageReturnLineFeed;
-      headersEndIndex = part.indexOf(splitter);
-      if (headersEndIndex < 0) {
-        splitter = "\n\n";
-        headersEndIndex = part.indexOf(splitter);
-        if (headersEndIndex < 0) {
-          throw new Error("No headers found in part.");
+export async function parseMtomResponse(
+  payload: Buffer,
+  boundary: string
+): Promise<IMTOMAttachments> {
+  return new Promise((resolve, reject) => {
+    const resp: IMTOMAttachments = {
+      parts: [],
+    };
+    let headerName = "";
+    let headerValue = "";
+    let data: Buffer;
+    let partIndex = 0;
+    const parser = new MultipartParser();
+
+    parser.initWithBoundary(boundary);
+    parser.on(
+      "data",
+      ({
+        name,
+        buffer,
+        start,
+        end,
+      }: {
+        name: string;
+        buffer: Buffer;
+        start: number;
+        end: number;
+      }) => {
+        switch (name) {
+          case "partBegin":
+            resp.parts[partIndex] = {
+              body: Buffer.from(""),
+              headers: {},
+            };
+            data = Buffer.from("");
+            break;
+          case "headerField":
+            headerName = buffer.slice(start, end).toString();
+            break;
+          case "headerValue":
+            headerValue = buffer.slice(start, end).toString();
+            break;
+          case "headerEnd":
+            //eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
can we get rid of this?

Reply:
tried a bunch of things but the typescript compiler still freaks even if I have this

`if (resp.parts[partIndex]) {
              resp.parts[partIndex].body = data;
              partIndex++;
            }`